walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/EventIterator.java

```diff
@@ -0,0 +1,132 @@
+// Copyright 2019 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https:// www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Comparator;
+
+public class EventIterator {
+
+  // filtered event lists sorted by start and end times
+  private List<TrimmedEvent> startSortedEvents;
+  private List<TrimmedEvent> endSortedEvents;
+
+  private int mandatoryEventAttendees;
+  private Set<String> optionalEventAttendees;
+
+  // The maximum value startIndex and endIndex can have 
+  private final int maxIndex;
+
+  // index of the current position in startSortedEvents and endSortedEvents
+  private int startIndex = 0;
+  private int endIndex = 0;
+
+  private TrimmedEvent currEvent;
+  private int currEventTime;
+
+  private boolean eventEndsNow;
+
+  public EventIterator(List<TrimmedEvent> events) {
+    maxIndex = events.size();
+    this.startSortedEvents =  sortEvents(events, TrimmedEvent.ORDER_BY_START);
```

Comment:
nit: use lambda for comparer could make it much less code, and easy to read to.
events.sort((a,b)->(a.start - b.start));
events.sort((a,b)->(a.end() - b.end()));